app/main/data_requests.py

```diff
@@ -24,10 +23,22 @@ def post_ingest(file: FileStorage, data: dict = None) -> Response:
     files = {"excel_file": (file.name, file, MIMETYPE.XLSX)}
 
     response = requests.post(request_url, files=files, data=data)
-
-    if response.status_code in [200, 400, 500]:
-        return response
-
-    else:
-        current_app.logger.error(f"Bad response: {request_url} returned {response.status_code}")
-        return abort(500)
+    response_json = response.json()
+
+    match response.status_code:
+        case 200:
+            return True, None, None
+        case 400:
+            if validation_errors := response_json.get("validation_errors"):
+                if pre_error := validation_errors.get("PreTransformationErrors"):
+                    return False, pre_error, None
+                elif tab_errors := validation_errors.get("TabErrors"):
+                    return False, None, tab_errors
+            # if json isn't as expected then 500
+            abort(500)
+        case 500:
+            current_app.logger.error(f"Ingest failed for an unknown reason - failure_id={response_json.get('id')}")
```

Comment:
This is where we log the failure_id via Submit. The rest of the changes are a refactor